src/Zend/Captcha/Image.php

```diff
@@ -496,10 +496,13 @@ protected function _generateImage($id, $word)
             $w = imagesx($img);
             $h = imagesy($img);
         }
-        $text_color = imagecolorallocate($img, 0, 0, 0);
-        $bg_color   = imagecolorallocate($img, 255, 255, 255);
+        $text_color = (int) imagecolorallocate($img, 0, 0, 0);
```

Comment:
All of these casts and the added exception throw were added to satisfy PHPStan. The casts are probably the existing behavior. The added exception does change behavior slightly in case of an error, but the warnings/errors that PHP would have thrown when trying to access array indexes on `false` would probably have resulted in a similar situation.